phoenix-core/src/it/java/org/apache/phoenix/end2end/EmptyColumnIT.java

```diff
@@ -63,851 +62,1006 @@
-		testCases.add(testCaseWhenAllCFMatchAndSame);
-
-		OtherOptions testCaseWhenAllCFMatch = new OtherOptions();
-		testCaseWhenAllCFMatch.setTestName("testCaseWhenAllCFMatch");
-		testCaseWhenAllCFMatch.setTableCFs(Lists.newArrayList(null, "A", "B"));
-		testCaseWhenAllCFMatch.setGlobalViewCFs(Lists.newArrayList(null, "A", "B"));
-		testCaseWhenAllCFMatch.setTenantViewCFs(Lists.newArrayList(null, "A", "B"));
-		testCases.add(testCaseWhenAllCFMatch);
-
-		OtherOptions testCaseWhenTableCFsAreDiff = new OtherOptions();
-		testCaseWhenTableCFsAreDiff.setTestName("testCaseWhenTableCFsAreDiff");
-		testCaseWhenTableCFsAreDiff.setTableCFs(Lists.newArrayList(null, "A", "B"));
-		testCaseWhenTableCFsAreDiff.setGlobalViewCFs(Lists.newArrayList("A", "A", "B"));
-		testCaseWhenTableCFsAreDiff.setTenantViewCFs(Lists.newArrayList("A", "A", "B"));
-		testCases.add(testCaseWhenTableCFsAreDiff);
-
-		OtherOptions testCaseWhenGlobalAndTenantCFsAreDiff = new OtherOptions();
-		testCaseWhenGlobalAndTenantCFsAreDiff.setTestName("testCaseWhenGlobalAndTenantCFsAreDiff");
-		testCaseWhenGlobalAndTenantCFsAreDiff.setTableCFs(Lists.newArrayList(null, "A", "B"));
-		testCaseWhenGlobalAndTenantCFsAreDiff.setGlobalViewCFs(Lists.newArrayList("A", "A", "A"));
-		testCaseWhenGlobalAndTenantCFsAreDiff.setTenantViewCFs(Lists.newArrayList("B", "B", "B"));
-		testCases.add(testCaseWhenGlobalAndTenantCFsAreDiff);
-
-		OtherOptions testCaseWhenNoDefaultCF = new OtherOptions();
-		testCaseWhenNoDefaultCF.setTestName("testCaseWhenNoDefaultCF");
-		testCaseWhenNoDefaultCF.setTableCFs(Lists.newArrayList("A", "A", "B"));
-		testCaseWhenNoDefaultCF.setGlobalViewCFs(Lists.newArrayList("A", "A", "B"));
-		testCaseWhenNoDefaultCF.setTenantViewCFs(Lists.newArrayList("A", "A", "B"));
-		testCases.add(testCaseWhenNoDefaultCF);
-
-		OtherOptions testCaseWhenNoDefaultCFAndGlobalAndTenantCFsAreDiff = new OtherOptions();
-		testCaseWhenNoDefaultCFAndGlobalAndTenantCFsAreDiff.setTestName("testCaseWhenNoDefaultCFAndGlobalAndTenantCFsAreDiff");
-		testCaseWhenNoDefaultCFAndGlobalAndTenantCFsAreDiff.setTableCFs(Lists.newArrayList("A", "A", "B"));
-		testCaseWhenNoDefaultCFAndGlobalAndTenantCFsAreDiff.setGlobalViewCFs(Lists.newArrayList("B", "B", "B"));
-		testCaseWhenNoDefaultCFAndGlobalAndTenantCFsAreDiff.setTenantViewCFs(Lists.newArrayList("A", "A", "A"));
-		testCases.add(testCaseWhenNoDefaultCFAndGlobalAndTenantCFsAreDiff);
+    private void assertAllHBaseRowsHaveEmptyColumnCell(byte[] hbaseTableName,
+            byte[] emptyColumnFamilyName, byte[] emptyColumnName, long minTimestamp,
+            int expectedRows) throws IOException, SQLException {
+
+        try (Table tbl = driver.getConnectionQueryServices(getUrl(), TestUtil.TEST_PROPERTIES)
+                .getTable(hbaseTableName)) {
+
+            Scan allRows = new Scan();
+            allRows.setMaxVersions(1);
+            allRows.setTimeRange(minTimestamp, minTimestamp + Integer.MAX_VALUE);
```

Comment:
I do not think we need to set the time range and max number of versions here. Scan by default always returns the latest version. Also, we need to check that empty column has the maximum timestamp value of a given row. I do not see this has been checked here.

Reply:
Will remove the maxVersion on Scan. 
The time range allows the test writer to specify the time range and the expected record count therein thus easier to specify the assert conditions.
I will add the check for max timestamp.